src/models/components/spiral_conv.py

```diff
@@ -0,0 +1,149 @@
+
+    def reset_hidden(self):
+        self.last_conv = None
+
+    def randomize_init(self):
+        self.last_conv_init.value = torch.randn(self.dim, dtype=torch.cfloat)
+
+    def set_is_refresh(self, is_refresh):
+        self.is_refresh = is_refresh
+
+    def get_hidden(self) -> Tensor:
+        return self.last_conv
+
+    def set_hidden(self, hidden: Tensor):
+        self.last_conv = hidden
+
+
+class ArchitectureBlock(nn.Module):
+    def __init__(self, dim: int, dim_ff_scale: float, dropout: float):
+        super().__init__()
+        self.spiral_conv_1 = SpiralConv(dim)
+        self.spiral_conv_2 = SpiralConv(dim)
+        self.ffn = FFN(dim, dim_ff_scale, dropout)
+        self.layer_norm = nn.LayerNorm(dim)
+        self.silu = nn.SiLU()
+        self.dropout = nn.Dropout(dropout)
+
+    def forward(self, x):
+        x_ = x
+        x = self.layer_norm(x)
+        x = self.spiral_conv_1(x)
+        x = self.silu(x)
+        x = self.spiral_conv_2(x)
+        x = self.dropout(x)
+        x = x + x_
+
+        x_ = x
+        x = self.layer_norm(x)
+        x = self.ffn(x)
+        x = x + x_
+
+        return x
+
+    def reset_hidden(self):
+        self.spiral_conv_1.reset_hidden()
+        self.spiral_conv_2.reset_hidden()
+
+    def randomize_init(self):
+        self.spiral_conv_1.randomize_init()
+        self.spiral_conv_2.randomize_init()
+
+    def set_is_refresh(self, is_refresh):
+        self.spiral_conv_1.set_is_refresh(is_refresh)
+        self.spiral_conv_2.set_is_refresh(is_refresh)
+
+    def get_hidden(self) -> (Tensor, Tensor):
+        return (self.spiral_conv_1.get_hidden(), self.spiral_conv_2.get_hidden())
+
+    def set_hidden(self, hidden: (Tensor, Tensor)):
+        hidden_1, hidden_2 = hidden
+        self.spiral_conv_1.set_hidden(hidden_1)
+        self.spiral_conv_2.set_hidden(hidden_2)
+
+
+class Architecture(nn.Module):
+    def __init__(self, depth: int, dim: int, dim_ff_scale: float, dropout: float):
+        super().__init__()
+        self.block_list = nn.ModuleList([ArchitectureBlock(dim, dim_ff_scale, dropout) for _ in range(depth)])
+
+    def forward(self, x):
+        for block in self.block_list:
+            x = block(x)
+        return x
+
+    def reset_hidden(self):
+        for block in self.block_list:
+            block.reset_hidden()
+
+    def randomize_init(self):
+        for block in self.block_list:
+            block.randomize_init()
+
+    def set_is_refresh(self, is_refresh):
+        for block in self.block_list:
+            block.set_is_refresh(is_refresh)
+
+    def get_hidden(self) -> [(Tensor, Tensor)]:
+        hidden_list = []
+        for block in self.block_list:
+            hidden_list.append(block.get_hidden())
+        return hidden_list
+
+    def set_hidden(self, hidden_list):
+        hidden_list = hidden_list.copy()
+        for block in self.block_list:
+            hidden = hidden_list.pop(0)
+            block.set_hidden(hidden)
```

Comment:
`pop(0)`はメモリの配置換えがおきるので高負荷です。よってアイテムアサインメントを使用してください。
```suggestion
        for idx, block in enumerate(self.block_list):
            block.set_hidden(hidden_list[i])
```

Reply:
修正しました https://github.com/MLShukai/PrimitiveAMI/pull/124/commits/94ba6a2536a6c64cdabc5bb32af5a980ce9557b5